cmd/snap/cmd_model.go

```diff
@@ -0,0 +1,229 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+)
+
+var (
+	shortModelHelp = i18n.G("Get the active model for this device")
+	longModelHelp  = i18n.G(`
+The model command returns only the active model assertion information for this 
+device.	
+
+The --serial option allows printing the serial assertion instead of the model
+assertion.
+
+The --verbose option prints off more headers from the assertion.
+
+The --assertion option prints off the raw assertion.
+`)
+
+	// this list is a "nice" "human" "readable" "ordering" of headers to print
+	// off, sorted in lexographical order with meta headers and primary key
+	// headers removed, and big nasty keys such as device-key-sha3-384 and
+	// device-key at the bottom
+	// it also contains both serial and model assertion headers, but we
+	// follow the same code path for both assertion types and some of the
+	// headers are shared between the two, so it still works out correctly
+	niceOrdering = []string{
+		"architecture",
+		"base",
+		"classic",
+		"display-name",
+		"gadget",
+		"kernel",
+		"revision",
+		"timestamp",
+		"required-snaps",
+		"device-key-sha3-384",
+		"device-key",
+	}
+)
+
+type cmdModel struct {
+	waitMixin
+	timeMixin
+	Serial    bool `long:"serial"`
+	Verbose   bool `long:"verbose"`
+	Assertion bool `long:"assertion"`
+}
+
+func init() {
+	addCommand("model",
+		shortModelHelp,
+		longModelHelp,
+		func() flags.Commander {
+			return &cmdModel{}
+		}, timeDescs.also(waitDescs.also(map[string]string{
+			"assertion": i18n.G("Print the raw assertion."),
+			"verbose":   i18n.G("Print more of the assertion fields."),
```

Comment:
As above, it would be good if we could qualify *more* with the headers that are additionally returned - is it all the headers, or how have they been selected?

Reply:
"Print all specific assertion fields" perhaps?

Reply:
That makes sense, I went with @pedronis's suggestion.